packages/rum-core/src/performance-monitoring/performance-monitoring.js

```diff
@@ -84,6 +86,42 @@ class PerformanceMonitoring {
     if (flags[FETCH]) {
       patchEventHandler.observe(FETCH, this.getFetchSub())
     }
+
+    if (flags[EVENT_TARGET]) {
+      patchEventHandler.observe(EVENT_TARGET, this.getEventTargetSub())
+    }
+  }
+
+  getEventTargetSub() {
+    const transactionService = this._transactionService
+    return (event, task) => {
+      if (
+        event === SCHEDULE &&
+        task.source === EVENT_TARGET &&
+        task.eventType === 'click'
+      ) {
+        let target = task.target
+        let name = target.getAttribute('name')
+        let classes = target.getAttribute('class')
```

Comment:
Using class might not be scalable as there are lots of conventions and if the websites uses css modules or something similar concept, those would be `autogenarated` from the bundler (Ex: webpack) and it wont help the user in any way. 

One idea would be to figure out where the element is present on the page, something like `xpath`. Example from github `/html/body/div[1]/header/div[5]/details/details-menu/div[3]/div/details/details-dialog/form/div[1]/button`, We can also use it in context than having it as long name.

Reply:
To be honest I only added the classes as a back up but not entirely happy about it because of the issues that you mentioned. But, I also don't think the xpath is good approach since it will clutter the UI. However, the classes are only a backup to the name attribute, so there's a way for users who can't use classes to fix their elements to have a name.